src/connectionAttempt.ts

```diff
@@ -0,0 +1,102 @@
+import { isCancelError, raceWithAbort } from '@mongodb-js/compass-utils';
+import type { ConnectionOptions, DataService } from 'mongodb-data-service';
+import { connect } from 'mongodb-data-service';
+
+import { createLogger } from './logging';
+import LINKS from './utils/links';
+
+// eslint-disable-next-line @typescript-eslint/no-var-requires
+const packageJSON = require('../package.json');
+
+const log = createLogger('connection attempt');
+
+function isConnectionAttemptTerminatedError(err: Error) {
+  return err?.name === 'MongoError' && err?.message === 'Topology closed';
```

Comment:
Is this the actual error message? The driver source code seems to imply that the current message is `Topology is closed` and has been for a while: https://github.com/mongodb/node-mongodb-native/blob/a4776cfe638c4e51cb690e43b941153d4f36a30a/src/error.ts#L783

I’d definitely check `err?.name === 'MongoTopologyClosedError'` here, either instead of or in addition to the `err?.message` check

Reply:
Updated. Not sure how to reproduce easily 😅